### PR TITLE
usm: map cleaner: Batch deletion fallback

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?
Added a fallback mechanism to a element by element deletion if the BatchDelete fails with KeyDoesNotExist error.

### Motivation
If a single key does not exist in the map, and we try to delete it using BatchDelete, the operation will fail with error KeyDoesNotExist, and we might partially delete the entries in the map.

### CODE REVIEW

File: pkg/ebpf/map_cleaner_test.go [hunk 1]
```diff
@@ -30,6 +30,59 @@ func TestMain(m *testing.M) {
...
+	// Clean all the even entries
+	cleaner.Clean(100*time.Millisecond, nil, nil, func(now int64, k int64, v int64) bool {
+		// Delete a random key.
+		if k == 4 {
+			m.Delete(&k)
+		}
+		return k%2 == 0
+	})
+
+	time.Sleep(1 * time.Second)
```
Comment (amitslavin) [inline pkg/ebpf/map_cleaner_test.go:69]: why the sleep is needed?

File: pkg/ebpf/map_cleaner.go [hunk 2]
```diff
@@ -137,6 +138,14 @@ func (mc *MapCleaner[K, V]) cleanWithBatches(nowTS int64, shouldClean func(nowTS
 	var deletionError error
 	if len(keysToDelete) > 0 {
 		deletedCount, deletionError = mc.emap.BatchDelete(keysToDelete)
+		if errors.Is(deletionError, ebpf.ErrKeyNotExist) {
```
Comment (amitslavin) [inline pkg/ebpf/map_cleaner.go:141]: please add documentation of the reason we want to add this code part

**Review by amitslavin — CHANGES_REQUESTED** [review]

File: pkg/ebpf/map_cleaner_test.go (hunk 1 above)
Comment (guyarb) [reply pkg/ebpf/map_cleaner_test.go:69]: giving some time to the cleaner to run

Comment (guyarb) [reply pkg/ebpf/map_cleaner_test.go:69]: added comment

**Review by amitslavin — APPROVED** [review]

**Pull request merged**